documentation/src/main/wiki/Adapter.md

```diff
@@ -1,7 +1,59 @@
-Adapters are used to communicate with external services. Knot.x recommends to create dedicated Adapter
-every time we need to perform some business logic or adapt service response to other format.
+# Adapters
+Adapter is a module that is responsible for communication between Knot.x (exactly [[Knots|Knot]]) and external services.
 
-# TODO
-- types of adapters
-- how to implement custom adapter
-- how to run custom adapter together with Knot.x
+[[assets/knotx-adapters.png|alt=Adapters]]
+
+
+## How does it work?
+Adapters are thought as project specific logic and we recommend to create dedicated Adapter
+every time some business logic or adapting service response to other format is required.
+
+Each Knot that its contract that adapter must meet. Find out more about [[Knot contracts|Knot#how-does-it-work]].
+
+
+### Types of adapters
+There are two types of Adapters that Knot.x core can communicate with:
+- [[Service Adapter|ServiceAdapter]] for [[Action Knot|ActionKnot]],
+- [[Action Adapter|ActionAdapter]] for [[View Knot|ViewKnot]].
+
+Knot.x comes with implementation of a [[Service Adapter|ServiceAdapter]], that enables communication with external 
+services using HTTP Protocol. See [[Http Service Adapter|HttpServiceAdapter]] for more information.
+Please note, that this implementation is very generic and we recommend to create project-specific 
+adapters for any custom solution.
+
+## How to configure?
+Adapter may have its configuration in form of JSON object entry. 
```

Comment:
Maybe we should provide here information about generic adapter configuration model: com.cognifide.knotx.adapter.api.AdapterConfiguration.